### PR TITLE
Drop unused `fs_extra` and `hamcrest2` dependencies

### PR DESCRIPTION
**Pull request by weirdan**
Fixes nushell/nushell#7995

# Description

This dependency is no longer used by nushell itself.

# User-Facing Changes

None.

# Tests + Formatting

Pased.

# After Submitting

None.

### CODE REVIEW

**Comment by sholderbach** [comment]
Good catch should be unused since https://github.com/nushell/nushell/pull/12022/

Wondering if #6583 is thus out of date as well

File: Cargo.lock [hunk 1]
```diff
@@ -6014,9 +6013,9 @@ dependencies = [
 
 [[package]]
 name = "sysinfo"
-version = "0.30.13"
+version = "0.30.11"
```
Comment (sholderbach) [inline Cargo.lock:6016]: Drunk `cargo` deciding to revert https://github.com/nushell/nushell/pull/13615

Comment (weirdan) [reply Cargo.lock:6016]: Yes, it did. I continue to be amazed by "user-friendliness" of rust ecosystem :imp:

Comment (weirdan) [reply Cargo.lock]: Fixed.

Comment (sholderbach) [reply Cargo.lock]: Thanks, that is so weird

**Pull request merged**